pkg/kubelet/container/sync_result.go

```diff
@@ -19,14 +19,59 @@ package container
 import (
 	"errors"
 	"fmt"
+	"time"
 
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 )
 
-// TODO(random-liu): We need to better organize runtime errors for introspection.
+// BackoffError should be used whenever an error needs to specify a particular backoff duration
+// to the Kubelet.
+type BackoffError struct {
+	error
+	backoffTime time.Time
+}
+
+func NewBackoffError(err error, backoffTime time.Time) *BackoffError {
+	return &BackoffError{
+		error:       err,
+		backoffTime: backoffTime,
+	}
+}
+
+// BackoffTime returns the expected expiration time of the backoff.
+func (e *BackoffError) BackoffTime() time.Time {
+	return e.backoffTime
+}
 
-// ErrCrashLoopBackOff returned when a container Terminated and Kubelet is backing off the restart.
-var ErrCrashLoopBackOff = errors.New("CrashLoopBackOff")
```

Comment:
is this string not used anywhere any longer? I see this string being used in the test in test/e2e_node/container_restart_test.go and I wonder if that string is coming from here.

Reply:
Ah, yeah, probably. I'll put the string back and return it appropriately.